src/System.Memory/src/System/Buffers/ReadOnlyBuffer_helpers.cs

```diff
@@ -180,14 +180,14 @@ private static SequencePosition SeekMultiSegment(IMemoryList<byte> start, int st
                 // if end is not trusted
                 if (!foundResult)
```

Comment:
Is there a reason we still have the foundResult flag? Was there an issue with the code I had suggested?

Reply:
Yes, I decided not to change main logic of the method in this pr.

Reply:
Fair enough. Can you please file an issue for it to optimize this method, especially if the tests pass with the changed logic?

Reply:
If we see an impact during perf investigation it would get optimized.

Reply:
That is fine. It can be deferred. Tbh, having boolean flags to control code flow makes it difficult to read this and reason about the behavior.